pkg/quota_management/account.go

```diff
@@ -1,20 +1,62 @@
 package quota_management
 
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+)
+
 type Account struct {
-	Username            string `yaml:"username"`
-	MaxAllowedInstances int    `yaml:"max_allowed_instances"`
+	Username            string    `yaml:"username"`
+	MaxAllowedInstances int       `yaml:"max_allowed_instances"`
+	GrantedQuota        QuotaList `yaml:"granted_quota,omitempty"`
+}
+
+var _ QuotaManagementListItem = &Account{}
+
+func (account Account) IsInstanceCountWithinLimit(instanceTypeID string, billingModelID string, count int) bool {
+	return count <= account.GetMaxAllowedInstances(instanceTypeID, billingModelID)
+}
+
+func (account Account) GetMaxAllowedInstances(instanceTypeID string, billingModelID string) int {
+	bm, ok := account.getBillingModel(instanceTypeID, billingModelID)
+
+	if !ok {
+		return 0
+	}
+
```

Comment:
I think we can change this a little bit to avoid some conditional nesting.
For example:

if bm.Allowed > 0 then return it

res := MaxAllowedInstances
if account.MaxAllowedInstances > 0 then res = account.MaxAllowedInstances
return res